pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -348,32 +350,59 @@ func (o *Options) Run() error {
 		}
 
 		if len(cv.Status.AvailableUpdates) > 0 {
-			fmt.Fprintf(o.Out, "Available Updates:\n\n")
-			w := tabwriter.NewWriter(o.Out, 0, 2, 1, ' ', 0)
-			fmt.Fprintf(w, "VERSION\tIMAGE\n")
+			fmt.Fprintf(o.Out, "\nRecommended updates:\n\n")
+			// set the minimal cell width to 14 to have a larger space between the columns for shorter versions
+			w := tabwriter.NewWriter(o.Out, 14, 2, 1, ' ', 0)
+			fmt.Fprintf(w, "  VERSION\tIMAGE\n")
 			// TODO: add metadata about version
-			sortSemanticVersions(cv.Status.AvailableUpdates)
+			sortReleasesBySemanticVersions(cv.Status.AvailableUpdates)
 			for _, update := range cv.Status.AvailableUpdates {
-				fmt.Fprintf(w, "%s\t%s\n", update.Version, update.Image)
+				fmt.Fprintf(w, "  %s\t%s\n", update.Version, update.Image)
 			}
 			w.Flush()
-			if c := findCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
+			if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 				fmt.Fprintf(o.ErrOut, "warning: Cannot refresh available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
 			}
 		} else {
-			if c := findCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
+			if c := findClusterOperatorStatusCondition(cv.Status.Conditions, configv1.RetrievedUpdates); c != nil && c.Status == configv1.ConditionFalse {
 				fmt.Fprintf(o.ErrOut, "warning: Cannot display available updates:\n  Reason: %s\n  Message: %s\n\n", c.Reason, strings.ReplaceAll(c.Message, "\n", "\n  "))
 			} else {
 				fmt.Fprintf(o.Out, "No updates available. You may force an upgrade to a specific release image, but doing so may not be supported and may result in downtime or data loss.\n")
 			}
 		}
 
+		if o.IncludeNotRecommended {
+			if containsNotRecommendedUpdate(cv.Status.ConditionalUpdates) {
+				sortConditionalUpdatesBySemanticVersions(cv.Status.ConditionalUpdates)
+				fmt.Fprintf(o.Out, "\nSupported but not recommended updates:\n\n")
```

Comment:
I think you want one less trailing `\n` here and one more below with `\n  Version...`, to give a blank line between each target-version block.

Reply:
Yes. I have forgotten I was testing with messages that were all ending with the character `\n`.

Is there any guarantee that the messages end or they do not end with the character `\n`?

Reply:
If we can't be sure,  I can add the [func TrimSuffix](https://pkg.go.dev/strings#TrimSuffix) for the `Message` and add the newlines as you have proposed.

Reply:
I have modified the new-lines as you have proposed and added `strings.ReplaceAll(strings.TrimSpace(c.Message), "\n", "\n  ")` when printing update's `Message`.

Fixed in https://github.com/openshift/oc/pull/961/commits/8b225ea2e172938fb0bf6e250d40185c920a85a2.